TypeCobol/Compiler/CupParser/NodeBuilder/CupParserTypeCobolProgramDiagnosticErrorReporter.cs

```diff
@@ -169,10 +169,18 @@ public class CupParserDiagnostic : Diagnostic
             // TO DO - IMPORTANT : this is the INITIAL line number, and not the CURRENT line number
             // This is enough to pass all unit tests, but will return false informations in real usage !
             // for real line number, use a Snapshot
-            if (Line < 0 && OffendingSymbol != null)
+            CodeElement e = offendingSymbol?.value as CodeElement;
```

Comment:
The idea here is to find accurate information for the parser error location, so a pseudo-code would be :
```csharp
if (Line < 0 && OffendingSymbol != null)
{
    if (offendingSymbol is inside copy)
    {
        //Retrieve copy token and use its line as location
    }
    else
    {
        //Fallback to first consumed token location
        ...
    }
}
```

To determine whether the offending symbol is inside a copy or not, you'll have to use two different methods because in ANTLR, we are dealing with tokens and in CUP we have codeElements so the code will be slightly different between the two classes.

Reply:
It's right.
Modification has been made in these 2 classes by removing code about line is < 0 because it never happens (result of property IToken.Line or CodeElement.Line).
Line is modified when a copy is detected.
https://github.com/TypeCobolTeam/TypeCobol/pull/1602/commits/c75e93cbbbe150f7b2561c1696660261a82adfbe